backend/app/models/operation.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# operations table
+class Operation < ActiveRecord::Base
+
+  def self.pending_operations(operation_ids)
+    sql = "select id, operation_type_id from operations where id in ( #{operation_ids.join(',')} ) and status = 'pending'"
+    operations = Operation.find_by_sql sql
+  end
+
+  def self.distinct_operation_types(operation_ids, status)
+    sql = "select distinct operation_type_id from operations where id in ( #{operation_ids.join(',')} ) and status = ?"
+    distinct = Operation.find_by_sql [sql, status]
+  end
+
+  def self.operations_for_category_type_status(category, type, status)
+    sql = "
+      select o.id, pa.plan_id, u.name, o.status, o.updated_at
+      from operation_types ot
+      inner join operations o on o.operation_type_id = ot.id
+      inner join plan_associations pa on pa.operation_id = o.id
+      inner join users u on u.id = o.user_id
+      where ot.category = ? and ot.name = ? and o.status = ?
+      order by o.updated_at desc
+    "
+    operations = Operation.find_by_sql [sql, category, type, status]
+  end
+
+  def self.set_status_for_ids(status, ids)
+    sql = "update operations set status = '#{status}' where id in ( #{ids.join(',')} )"
+    Operation.connection.execute sql
+  end
+
+  def self.set_status_for_job(status, job_id)
+    sql = "update operations set status = '#{status}' where id in ( select operation_id from job_associations where job_id = #{job_id} )"
+    Operation.connection.execute sql
+  end
+
+  def self.operation_from_job(operation_id, job_id)
```

Comment:
huh?  This method is a validation method to verify the right thing is vetted before it is attempted.

When the user removes an operation from a job, we call this method first to retrieve the operation_id and make sure it is in the job_id before attempting to remove it.  We could eliminate this extra check and verify things downstream, but I like catching things earlier rather than later in the process 

See jobs_controller > def remove. The call that removes the operation from the job is JobAssociation.remove_operation_from_job(operation_id, job.id).  This call comes before that.

Reply:
done. (At the end of the day this is a "nice-to-have" check that isn't required, and probably cleaner to leave it out.)